### DISCUSSION TITLE
Change Bevy Text component behavoir so it does not need to be a child node

### DISCUSSION THREAD

**Discussion by abvanliew** [bevyengine/bevy#16774] [General]
I have been learning the new UI components in Bevy 0.15. I found the text component to be counter intuitive. For example when I was trying to make a simple UI element with centered text. I constructed the following entity. I expect that the text would show up centered on the UI element. 

```
commands.spawn( (
    Node {
      justify_content: JustifyContent::Center,
      align_items: AlignItems::Center,
      height: Val::Px( 50.0 ),
      width: Val::Px( 100.0 ),
      ..Default::default()
    },
    Text( "A".into() ),
    BackgroundColor( Color::BLACK ),
    BorderRadius::all( Val::Px( 4.0 ) ),
  ) )
```

However I found that what I needed to do was separate out the text into a child entity of the first element with the following code.

```
  let tile_b = commands.spawn( (
    Node {
      justify_content: JustifyContent::Center,
      align_items: AlignItems::Center,
      height: Val::Px( 50.0 ),
      width: Val::Px( 100.0 ),
      ..Default::default()
    },
    BackgroundColor( Color::BLACK ),
    BorderRadius::all( Val::Px( 4.0 ) ),
  ) ).id();

  let tile_text_b = commands.spawn( Text( "B".into() ) ).id();
  commands.entity( tile_text_b ).set_parent( tile_b );
```

![bevy_text_alignment](https://github.com/user-attachments/assets/a3018b15-4396-47e5-850b-f947fc2a9d93)

For simple UI elements like buttons, I would expect that the Text component would be part of the same entity where I would control the color of the border or background for simple hover and button press logic. Would it be possible to refactor the Text component to work that way? Or is there a way to get the Text component to align without making a separate entity?

**Comment by Liam19** [comment]
I'm having a similar problem with a Label component for my game entities. I can't see a way to position the label or make it draw on top of the entity's sprite
